File: Python/program/battery.py
class Battery:
    def __init__(self, capacity, battery_life):
        self._capacity = capacity  # Kapasitas dalam mAh atau Wh
        self._battery_life = battery_life  # Daya tahan dalam jam

    def get_capacity(self):
        return self._capacity

    def set_capacity(self, capacity):
        self._capacity = capacity

    def get_battery_life(self):
        return self._battery_life

    def set_battery_life(self, battery_life):
        self._battery_life = battery_life

    def display_battery_info(self):
        print(f"Battery Capacity: {self._capacity} mAh, Battery Life: {self._battery_life} hours")

File: CPP/program/Battery.cpp
#pragma once
#include <iostream>

class Battery {
private:
    int capacity;
    int batteryLife;
public:
    Battery(int capacity, int batteryLife) : capacity(capacity), batteryLife(batteryLife) {}
    ~Battery() { std::cout << "Battery destroyed\n"; }
    int getCapacity() const { return capacity; }
    void setCapacity(int capacity) { this->capacity = capacity; }
    int getBatteryLife() const { return batteryLife; }
    void setBatteryLife(int batteryLife) { this->batteryLife = batteryLife; }
    void displayBatteryInfo() const {
        std::cout << "Battery Capacity: " << capacity << " mAh, Battery Life: " << batteryLife << " hours" << std::endl;
    }
};